source/MRViewer/MRRibbonMenu.cpp

```diff
@@ -91,7 +91,7 @@ void RibbonMenu::init( MR::Viewer* _viewer )
     callback_draw_custom_window = [&] ()
     {
         prevFrameObjectsCache_ = selectedObjectsCache_;
-        selectedObjectsCache_ = getAllObjectsInTree<const Object>( &SceneRoot::get(), ObjectSelectivityType::Selected );
+        selectedObjectsCache_ = SceneCache::getAllObjects<const Object, ObjectSelectivityType::Selected>();
```

Comment:
do we need local cache?

Reply:
it need for update if selection changed

Reply:
we can pass `SceneCache::getAllObjects<const Object, ObjectSelectivityType::Selected>()` it there